### ISSUE TITLE
MySQL 8 caching_sha2_authetication.so missing?

### ISSUE THREAD

**Issue by vierlex** [docksal/service-db#19]
I am having trouble being able to run MySQL 8
It took me a while to figure out any error messages (https://github.com/drush-ops/drush/issues/4256)

But finally I got this:
> ERROR 1045 (28000): Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such > file or directory

which led me to https://www.drupal.org/project/drupal/issues/2966523 which led me to adding a custom my.cnf under .docksal/etc/mysql/my.cnf with the contents:
> [mysqld]
> default_authentication_plugin=mysql_native_password

Which will lead to this error whenever I visit the site or execute some drush sql:* commands
>ERROR 1044 (42000): Access denied for user 'user'@'%' to database 'drupal'

My host is 'db' though, I am clueless where that % is coming from.
Maybe because of the changed authentication method..?
All the other settings (user,pw,port) are correctly set though.

**Comment by achekulaev** [comment]
1. Please describe steps to reproduce
    1. ...
    2. ...
    3. ...
2. Describe expected result
3. Describe Actual Result

**Comment by ChristianAdamski** [comment]
I think I can chip in here.

Using docksal/db:edge-mysql-8.0 and docksal/cli:edge, same issue as above.

I think the issue is not the DB-image, but the CLI one. It uses 

docker@cli:/var/www$ mysql --version
mysql  Ver 15.1 Distrib 10.3.22-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

and then failes to connect to the MySQL server with the above missing plugin

**Comment by ChristianAdamski** [comment]
It's enough to simply type "fin bash" and "mysql" to get

docker@cli:/var/www$ mysql
ERROR 1045 (28000): Plugin caching_sha2_password could not be loaded: /usr/lib/x86_64-linux-gnu/mariadb19/plugin/caching_sha2_password.so: cannot open shared object file: No such file or directory

**Comment by ChristianAdamski** [comment]
I might add, the entire directory does not exist " /usr/lib/x86_64-linux-gnu/mariadb19"

**Comment by lmakarov** [comment]
Pushed a fix for the 8.0 MySQL image (#22)

It's now available on the edge tag (`docksal/mysql:8.0-edge`).

```
$ fin db cli 'select version();'
mysql: [Warning] Using a password on the command line interface can be insecure.
+-----------+
| version() |
+-----------+
| 8.0.20    |
+-----------+
```
